### ISSUE TITLE
Handle OSC for setting 256-color values

### ISSUE THREAD

**Issue by jwilm** [jwilm/alacritty#26]
#24 added support for parsing and displaying them, now the OSC commands for setting the colors should be handled as well.

**Comment by jonhoo** [comment]
Without this, `vim` looks really silly with 256-bit color themes that I use with other terminals (like `base16`):
![256-bit vim color theme in alacritty](https://cloud.githubusercontent.com/assets/176295/22608670/e227a57a-ea2b-11e6-9653-6e2f131c2c39.png)

**Comment by jwilm** [comment]
This is really easy to add, I'll work on it this evening.

**Comment by jonhoo** [comment]
ping @jwilm: this is basically the only thing keeping me from using alacritty in my day-to-day

**Comment by jwilm** [comment]
@jonhoo I've actually started on this, it's a bit more work than I anticipated. Will probably have something by the weekend.

**Comment by jonhoo** [comment]
@jwilm thanks! much appreciated!

**Comment by jwilm** [comment]
This should be good to go. Please let me know if you run into any problems!

**Comment by jonhoo** [comment]
@jwilm hmm, how would I go about setting these in the alacritty config? It's not immediately obvious to me from looking at the diff.

**Comment by jwilm** [comment]
Oh, there's no way to do this from config. You might want to add it to your startup script. Something like

```
printf "\x1b]4;%d;rgb:%2.2x/%2.2x/%2.2x\x1b\\" 10 255 255 255
```

`10` is the index, and the next 3 values are R, G, and B.

**Comment by jwilm** [comment]
How did this work in your previous terminal?

**Comment by jonhoo** [comment]
With URxvt, this works using `*.color16`, `*.color17`, etc. in `.Xresources`. I'm fine setting it using escape codes, though that feels strange given how colors can already be set using `colors.normal` and `colors.bright` in `.config/alacritty.yml`.

**Comment by jwilm** [comment]
ahh, sure. That sounds like it would make sense to add. Very low on my priority list, though.

**Comment by jonhoo** [comment]
For what it's worth, the fix in 2be3e8fc5b72698391423c5b10cf490211a8828a completely fixes my use-case, because I also source color files in my shell, which set all the colors appropriately. This would only matter for cases when the terminal directly executes some non-shell command, but that's pretty rare (at least for me).

**Comment by gmile** [comment]
@jonhoo can you elaborate a little on how did you manage to fix the issue after all? I'm seeing pretty much the same issues with colors in vim right now – same as on your [screenshot](https://github.com/jwilm/alacritty/issues/26#issuecomment-277364073).

Did you have to run something from the terminal to fix the colors?
